my_enumerable.rb

```diff
@@ -1,85 +1,167 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength
 module Enumerable
   def my_each
-    return to_enum(_method_) unless block_given?
+    return to_enum(__method__) unless block_given?
 
+    array = to_a
     i = 0
-    until i == length
-      yield(self[i])
+    until i == array.length
+      yield(array[i])
       i += 1
     end
+    if (array.is_a?(Numeric) || array.is_a?(String)) && Range.new(array.first, array.last) == self
+      self
+    else
+      array
+    end
   end
```

Comment:
- [ ] When a block is given and `my_each` is called on a range, it should return the range itself after calling the given block once for each element.

```ruby
block = proc { |num| num < 4 }
range = Range.new(5, 50) 
range.my_each(&block) === range.each(&block) => false
```

_Hint: Yours is currently returning all of the elements in the range and not the range._